wallet/src/response.rs

```diff
@@ -0,0 +1,9 @@
+//! Utilities related to JsonRPC responses.
+
+use futures::Future;
+use jsonrpc_core::Error;
+
+/// TODO: doc
+pub trait Response<I>: Future<Item = I, Error = Error> {}
```

Comment:
Good idea! However, keep in mind that the `fn a() -> impl Response<T>` syntax cannot be used when the function returns different types of futures. In that case, we need boxed futures as defined in `node/src/actors/jsonrpc/json_rpc_methods.rs`:

```
type JsonRpcResultAsync<T> = Box<dyn Future<Item = T, Error = jsonrpc_core::Error> + Send>;
```

Reply:
I believe the Box is not needed since you can always combine different futures into one of the same type.

Reply:
Oh, I see, that's what the `Either` type is for: https://tokio.rs/docs/futures/combinators/#returning-from-multiple-branches